Bottom: b4db52ff5ea3c236b6b952a4c9d74bd00fd16a58
Top:    713af3ce6d219f968b4f710999b1063d6a70868b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-05 16:42:18 +0100

hid/gtk: Don't access a GtkMenuShell's children directly

In GTK3.0, direct access to shell->children will be impossible.


---

diff --git a/src/hid/gtk/ghid-main-menu.c b/src/hid/gtk/ghid-main-menu.c
index 9ac0367..2fe0a8b 100644
--- a/src/hid/gtk/ghid-main-menu.c
+++ b/src/hid/gtk/ghid-main-menu.c
@@ -343,20 +343,27 @@ ghid_main_menu_real_add_resource (GHidMainMenu *menu, GtkMenuShell *shell,
           /* If we get here, the resource is "-" or "@foo" for some foo */
           if (res->v[i].value[0] == '@')
             {
+              GList *children;
+              int pos;
+
+              children = gtk_container_get_children (GTK_CONTAINER (shell));
+              pos = g_list_length (shell->children);
+              g_list_free (children);
+
               if (strcmp (res->v[i].value, "@layerview") == 0)
                 {
                   menu->layer_view_shell = shell;
-                  menu->layer_view_pos = g_list_length (shell->children);
+                  menu->layer_view_pos = pos;
                 }
               else if (strcmp (res->v[i].value, "@layerpick") == 0)
                 {
                   menu->layer_pick_shell = shell;
-                  menu->layer_pick_pos = g_list_length (shell->children);
+                  menu->layer_pick_pos = pos;
                 }
               else if (strcmp (res->v[i].value, "@routestyles") == 0)
                 {
                   menu->route_style_shell = shell;
-                  menu->route_style_pos = g_list_length (shell->children);
+                  menu->route_style_pos = pos;
                 }
               else
                 Message (_("GTK GUI currently ignores \"%s\" in the menu\n"
